plugins/modules/ec2_vpc_net.py

```diff
@@ -616,19 +573,80 @@ def wait_for_updates(connection, module, vpc_id, ipv6_cidr, expected_cidrs, dns_
     if tags is not None:
         tag_list = ansible_dict_to_boto3_tag_list(tags)
         filters = [{"Name": f"tag:{t['Key']}", "Values": [t["Value"]]} for t in tag_list]
-        wait_for_vpc(module, connection, VpcIds=[vpc_id], Filters=filters)
+        wait_for_vpc(module, connection, waiter_name="vpc_available", VpcIds=[vpc_id], Filters=filters)
 
     wait_for_vpc_attribute(connection, module, vpc_id, "enableDnsSupport", dns_support)
     wait_for_vpc_attribute(connection, module, vpc_id, "enableDnsHostnames", dns_hostnames)
 
     if dhcp_id is not None:
         # Wait for DhcpOptionsId to be updated
         filters = [{"Name": "dhcp-options-id", "Values": [dhcp_id]}]
-        wait_for_vpc(module, connection, VpcIds=[vpc_id], Filters=filters)
+        wait_for_vpc(module, connection, waiter_name="vpc_available", VpcIds=[vpc_id], Filters=filters)
 
     return
 
 
+def ensure_present(connection, module: AnsibleAWSModule, vpc_id: str) -> None:
+    name = module.params.get("name")
+    cidr_block = module.params.get("cidr_block")
+    ipv6_cidr = module.params.get("ipv6_cidr")
+    purge_cidrs = module.params.get("purge_cidrs")
+    tenancy = module.params.get("tenancy")
+    dns_support = module.params.get("dns_support")
+    dns_hostnames = module.params.get("dns_hostnames")
+    dhcp_id = module.params.get("dhcp_opts_id")
+    tags = module.params.get("tags")
+    purge_tags = module.params.get("purge_tags")
+
+    if vpc_id is None:
+        if name is None:
+            module.fail_json("The name parameter must be specified when creating a new VPC.")
+        vpc_id = create_vpc_net(connection, module, cidr_block[0], tenancy, tags, ipv6_cidr, name)
+        changed = True
+        vpc_obj = get_vpc(module, connection, vpc_id)
+        if len(cidr_block) > 1:
+            cidrs_changed, desired_cidrs = update_cidrs(connection, module, vpc_obj, vpc_id, cidr_block, purge_cidrs)
+            changed |= cidrs_changed
+        else:
+            desired_cidrs = None
+        # Set on-creation defaults
+        if dns_hostnames is None:
+            dns_hostnames = True
+        if dns_support is None:
+            dns_support = True
+    else:
+        changed = False
+        vpc_obj = get_vpc(module, connection, vpc_id)
+        cidrs_changed, desired_cidrs = update_cidrs(connection, module, vpc_obj, vpc_id, cidr_block, purge_cidrs)
+        changed |= cidrs_changed
+        changed |= update_ipv6_cidrs(connection, module, vpc_obj, vpc_id, ipv6_cidr)
+        changed |= update_vpc_tags(connection, module, vpc_id, tags, name, purge_tags)
```

Comment:
Will something like this work?
```suggestion
    if vpc_id is None:
        if name is None:
            module.fail_json("The name parameter must be specified when creating a new VPC.")
        vpc_id = create_vpc_net(connection, module, cidr_block[0], tenancy, tags, ipv6_cidr, name)
        changed = True
        vpc_obj = get_vpc(module, connection, vpc_id)
        # Set on-creation defaults
        if dns_hostnames is None:
            dns_hostnames = True
        if dns_support is None:
            dns_support = True
    else:
        changed = False
        vpc_obj = get_vpc(module, connection, vpc_id)
        changed |= update_ipv6_cidrs(connection, module, vpc_obj, vpc_id, ipv6_cidr)
        changed |= update_vpc_tags(connection, module, vpc_id, tags, name, purge_tags)

    cidrs_changed, desired_cidrs = update_cidrs(connection, module, vpc_obj, vpc_id, cidr_block, purge_cidrs)
    changed |= cidrs_changed
```